torch/distributed/fsdp/_optim_utils.py

```diff
@@ -132,6 +132,12 @@ def _unflatten_optim_state(
         if to_save or shard_state
         else []
     )
+    if to_save:
+        for optim_state in unflat_param_state:
+            for key in list(optim_state.keys()):
```

Comment:
Same as the previous PR: I wonder why we do `list(optim_state.keys())`?

Reply:
If we use optim_state.items() and modify the dictionary in-place, python will complain. Even though using keys() won't have this issue, I am used to make it a list to avoid the confusion (not iterating and modifying the same data structure). I can remove this if that causes confusion.

Reply:
No worries. I was just curious.